src/sentry/profiles/task.py

```diff
@@ -206,59 +209,47 @@ def _symbolicate(
-        else:
-            _process_symbolicator_results_for_rust(profile, stacktraces)
-    elif profile["platform"] == "cocoa" and "version" in profile.get("profile", {}):
-        _process_symbolicator_results_for_cocoa_sample(profile, stacktraces)
-
-    if "version" in profile.get("profile", {}):
-        profile["profile"]["frames"] = stacktraces[0]["frames"]
+        _process_symbolicator_results_for_rust(profile, stacktraces)
 
-    # remove debug information we don't need anymore
-    profile.pop("debug_meta")
+    # rename the profile key to suggest it has been processed
+    profile["profile"] = profile.pop("sampled_profile")
 
-    if "version" not in profile["profile"]:
-        # rename the profile key to suggest it has been processed
-        profile["profile"] = profile.pop("sampled_profile")
 
+def _process_symbolicator_results_for_sample(profile: Profile, stacktraces: List[Any]):
+    if profile["platform"] == "rust":
+        for frame in stacktraces[0]["frames"]:
+            frame.pop("pre_context", None)
+            frame.pop("context_line", None)
+            frame.pop("post_context", None)
 
-def _process_symbolicator_results_for_cocoa_sample(profile: Profile, stacktraces: List[Any]):
-    for sample in profile["profile"]["samples"]:
-        stack = profile["profile"]["stacks"][sample["stack_id"]]
-
-        if len(stack) < 2:
-            continue
-
-        frame = profile["profile"]["frames"][stack[-1]]
+        def truncate_stack_needed(frame):
+            return frame.get("function", "") == "perf_signal_handler"
 
-        # here we exclude the frames related to the profiler itself as we don't care to profile the profiler.
-        if frame.get("instruction_addr", "") == "0xffffffffc":
-            stack = stack[:2]
+    elif profile["platform"] == "cocoa":
 
+        def truncate_stack_needed(frame):
+            return frame.get("instruction_addr", "") == "0xffffffffc"
 
-def _process_symbolicator_results_for_rust_sample(profile: Profile, stacktraces: List[Any]):
-    for frame in stacktraces[0]["frames"]:
-        frame.pop("pre_context", None)
-        frame.pop("context_line", None)
-        frame.pop("post_context", None)
+    else:
 
-    profile["profile"]["frames"] = stacktraces[0]["frames"]
+        def truncate_stack_needed(frame):
+            return False
 
     for sample in profile["profile"]["samples"]:
-        stack = profile["profile"]["stacks"][sample["stack_id"]]
+        stack_id = sample["stack_id"]
+        stack = profile["profile"]["stacks"][stack_id]
 
         if len(stack) < 2:
             continue
 
         frame = profile["profile"]["frames"][stack[-1]]
 
-        # here we exclude the frames related to the profiler itself as we don't care to profile the profiler.
-        if frame.get("function", "") == "perf_signal_handler":
-            stack = stack[:2]
+        if truncate_stack_needed(frame):
+            profile["profile"]["stacks"][stack_id] = stack[:-2]
```

Comment:
can we add a quick comment here explaining the `stack[:-2]`, this is non obvious for anyone just reading this file without the necessary context

Reply:
Also, will we be doing this in python as well? Since the resulting profile will have one extra frame for the sampling function either on the main thread or on the sampling thread (undecided yet).

Reply:
We can yes. I'm not sure what we have in `vroom` right now for Python.